### ISSUE TITLE
[Debugging Trick] Use Build Scheme ENAScreenshots with specific Launch Arguments to test the user interface on the simulator

### ISSUE THREAD

**Issue by ndegendogo** [corona-warn-app/cwa-app-ios#3804]
* [x] None of the other issue templates apply to this issue.

### Problem:
With the simulator you are able to test and debug a lot of the cwa code.
However, a lot of code pathes are not easy to cover.
Examples are: What happens after a green or a red warning? After a red antigen test result? After a red PCR test result?

If you are mainly interested in testing and debugging of the **user interface**, the following trick may help you.
Let's start with an example: You are interested to test the user interface after a green warning.

### Recipe for this specific use case:
* in Xcode, select scheme **ENAScreenshots** (instead of ENACommunity), and **Build for Testing**

<img width="877" alt="Screenshot 2021-11-07 at 20 20 27" src="https://user-images.githubusercontent.com/6118267/140660083-3e54899e-5acf-4a92-b585-e5164c531b16.png">

* open file ENAUITests/Home/ENAUITests_01a_Home
* search function test_homescreen_riskCardLow_riskMultipleDays
  The implementation sets several launch arguments, followed by the line `app.launch()`
* set a breakpoint after this line `app.launch()`

<img width="877" alt="Screenshot 2021-11-07 at 20 28 18" src="https://user-images.githubusercontent.com/6118267/140660310-602d210f-5185-4624-9c1b-fa39c5a1c88d.png">

* click the small triangle at the function definition to start the test, and wait till the breakpoint hits and the simulator device has launched
* you see the green risk card, showing exposures with low risk on 4 days (according to the launch arguments).
But the amazing part is: you can now take over, and manually test what you want (e.g. have a look at the explanations for the risk card, at the contact journal, etc)

**Note:** the supported launch arguments are defined in ENAUITests/LaunchArguments.swift

### Do not forget:
***The Scheme ENAScreenshots are configured with a focus to test the user interface.
Most of the processing "below the surface" is mocked, so you will not be able to debug the true code here.***

**Comment by whiskey** [comment]
If you want to explore a little more freely, you can also create a custom Scheme (which is **not** shared) to trigger some of the mocks.

![Screenshot 2021-11-08 at 16 44 04](https://user-images.githubusercontent.com/149976/140772960-449a0b79-01b3-4541-acdb-a8f0efa6e65d.png)

**Comment by ndegendogo** [comment]
Thanks for sharing @whiskey - such tricks are always useful 👍😅😎🎉

**Comment by Ein-Tim** [comment]
Hey @whiskey Do you happen to know if I can also "fake" having a positive test result in the app using this feature (launch arguments)?

I want to test https://github.com/corona-warn-app/cwa-app-ios/pull/4331.

**Comment by ndegendogo** [comment]
@Ein-Tim I can check in the evening after work

**Comment by Ein-Tim** [comment]
@ndegendogo 

That would be great! You can also write me a DM via Slack, if that's easier for you :D

**Comment by ndegendogo** [comment]
> if I can also "fake" having a positive test result in the app

Yes. Try my original procedure with [this test](https://github.com/corona-warn-app/cwa-app-ios/blob/64ab60e65b3bcfe2450f6766c6b23e80dfe7f46f/src/xcode/ENA/ENAUITests/ExposureSubmission/ENAUITests_04a_ExposureSubmission.swift#L358)
